internal/venice-client-common/src/main/java/com/linkedin/venice/stats/metrics/MetricEntityStateTwoEnums.java

```diff
@@ -0,0 +1,179 @@
+package com.linkedin.venice.stats.metrics;
+
+import static com.linkedin.venice.stats.VeniceOpenTelemetryMetricsRepository.REDUNDANT_LOG_FILTER;
+
+import com.linkedin.venice.stats.VeniceOpenTelemetryMetricsRepository;
+import com.linkedin.venice.stats.dimensions.VeniceDimensionInterface;
+import com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions;
+import io.opentelemetry.api.common.Attributes;
+import io.tehuti.metrics.MeasurableStat;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Similar to {@link MetricEntityStateOneEnum} but with two dynamic dimensions and 2 level EnumMap
+ */
+public class MetricEntityStateTwoEnums<E1 extends Enum<E1> & VeniceDimensionInterface, E2 extends Enum<E2> & VeniceDimensionInterface>
```

Comment:
That's interesting! I had never seen `&` used in an `extends` clause before...!

Reply:
This is where I ended up to have `getDimensionName()` and `getDimensionValue()` on every enums via implemeting `VeniceDimensionInterface` and to be able to call these functions inside this class on the enum classes passed in 😄